llvm/lib/Transforms/InstCombine/InstCombineMulDivRem.cpp

```diff
@@ -349,7 +349,7 @@ Instruction *InstCombinerImpl::visitMul(BinaryOperator &I) {
 
   // -X * Y --> -(X * Y)
   // X * -Y --> -(X * Y)
-  if (match(&I, m_c_Mul(m_OneUse(m_Neg(m_Value(X))), m_Value(Y))))
```

Comment:
Think this will cause regressions if the `-x` is not used by a subsequent `mul`, but by some other independent instruction.

Maybe match the full pattern?